models/articles-model.js

```diff
@@ -27,13 +27,28 @@ exports.selectArticleById = (article_id) => {
   return db
     .query(
       `SELECT * FROM articles
-    WHERE article_id = $1;`,
+      WHERE article_id = $1;`,
       [article_id]
     )
     .then(({ rows }) => {
-      if (rows.length === 0) {
+      if (rows.length === 0)
+        return Promise.reject({ msg: "Resource not found", code: 404 });
+      return rows[0];
+    });
+};
+
+exports.updateArticle = (article_id, inc_votes) => {
+  return db
+    .query(
+      `UPDATE articles
+      SET votes = votes + $1 
+      WHERE article_id = $2
+      RETURNING *`,
+      [inc_votes, article_id]
+    )
+    .then(({ rows }) => {
```

Comment:
We need to check whether the article exists or not in order to determined the reason a patch may fail.If it does not exist, it should throw a 404